src/main/java/Duke.java

```diff
@@ -1,10 +1,402 @@
+        }
+    }
+
+
+    /**
+     * Creates new DeadlineTask.
+     * Prints error message if invalid.
+     *
+     * @param input Input value by user.
+     */
+    private static void addDeadline(String input) {
+        try {
+            if (!input.contains(" /by ")) {
+                throw new MissingKeywordException();
+            }
+            String[] inputSplit = input.substring(LENGTH_OF_WORD_DEADLINE + 1).split(" /by ");
+            if (taskCount >= MAX_SIZE) {
+                //Array full
+                throw new ListFullException();
+            } else if (inputSplit.length < 2) {
+                //Invalid input
+                throw new DescriptionSplitException();
+            }
+            tasks.add(new DeadlineTask(inputSplit[0], inputSplit[1]));
+            taskCount++;
+            printAddedContent(inputSplit[0]);
+        } catch (ListFullException e) {
+            System.out.println(MESSAGE_LIST_FULL);
+        } catch (MissingKeywordException e) {
+            System.out.println(MESSAGE_MISSING_BY_KEYWORD);
+        } catch (Exception e) {
+            System.out.println(MESSAGE_UNRECOGNIZED_COMMAND);
+        }
+    }
+
+    /**
+     * Deletes multiple tasks from list.
+     *
+     * @param input Input value by user.
+     */
+    public static void deleteTasks(String input) {
+        try {
+            String[] indices = input.split(" ");
+            if (indices.length < 2) {
+                throw new DescriptionSplitException();
+            }
+            TreeSet<Integer> validIndices = new TreeSet<>();
+            for (int i = 1; i < indices.length; i++) {
+                int index = processIndex(indices[i]);
+                if (index != -1) {
+                    validIndices.add(index);
+                }
+            }
+            if (validIndices.size() < 1) {
+                throw new IndexOutOfBoundsException();
+            }
+            NavigableSet<Integer> sortedIndices = validIndices.descendingSet();
+            System.out.println(MESSAGE_DELETED);
+            for (int index : sortedIndices) {
+                tasks.get(index).printStatus();
+                System.out.println();
+                tasks.remove(index);
+                taskCount--;
+            }
+            System.out.println(LINE);
+        } catch (Exception e) {
+            System.out.println(MESSAGE_UNRECOGNIZED_COMMAND);
+        }
+    }
+
+    /**
+     * Processes index task from list. Returns -1 if invalid.
+     *
+     * @param stringIndex Index of task to remove.
+     */
+    public static int processIndex(String stringIndex) {
+        try {
+            int intIndex = Integer.parseInt(stringIndex) - 1;
+            if (intIndex >= taskCount || intIndex < 0) {
+                //Out of bounds
+                throw new IndexOutOfBoundsException();
+            }
+            return intIndex;
+        } catch (Exception e) {
+            return -1;
```

Comment:
Avoid magic number. In this case, it'd be better if you declare `-1` as a constant and name it appropriately and use that same constant in `deleteTask` method for better readability